ui/shared/components/panel/variants/VariantIndividuals.jsx

```diff
@@ -377,6 +397,65 @@ const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById })
           {filters}
         </small>
       )}
+    </div>
+  )
+}
+
+GenotypeQuality.propTypes = {
+  genotype: PropTypes.object,
+  variant: PropTypes.object,
+  showSampleType: PropTypes.bool,
+}
+
+const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById }) => {
+  const individualGenotypes = variant.genotypes[individual.individualGuid]
+  if (!individualGenotypes) {
+    return null
+  }
+  const genotypes = (Array.isArray(individualGenotypes) ? individualGenotypes : [individualGenotypes]).sort(
+    (a, b) => SAMPLE_TYPE_DISPLAY_ORDER.indexOf(a.sampleType) - SAMPLE_TYPE_DISPLAY_ORDER.indexOf(b.sampleType),
+  )
+
+  if (!isCalled(genotypes[0].cn) && !isCalled(genotypes[0].numAlt)) {
+    return <b>NO CALL</b>
+  }
+
+  if (genotypes[0].otherSample) {
+    genotypes.push(genotypes[0].otherSample)
+  }
+
+  const isHemiX = isHemiXVariant(variant, individual)
+
+  const sampleTypeWarnings = genotypes.reduce((acc, genotype) => {
+    const warnings = getWarningsForGenotype(genotype, variant, individual, isHemiX, isCompoundHet)
+    warnings.forEach((warning) => {
```

Comment:
this reduction is not used in the case where theres only one sample type, so it would be neater to just do a `genotypes.map` here to get an array of warnings. Then for the single sample case your warnings would just be `genotypeWarnings[0].join('. ')` and in `getWarningsForGenotypes` you could do this grouping, 
something like
```
`genotypeWarnings.reduce((acc, [warnings, index]) => {
  const { sampleType } = genotypes[index]
  warnings.forEach((warning) => {
    ...
}, {})
```

Reply:
agree, done!